File: LeetCode/MaximumDepthOfBinaryTree/maxDepthOfBT.c
#include <stdio.h>
#include <malloc.h>

typedef struct Node {
    int val;
    struct Node *left;
    struct Node *right;
}TNode, *TreeNode;


TreeNode createTree();
int maxDepth(TreeNode root);

int main(int argc, const char *argv[])
{
    TreeNode T;
    T = createTree();
    printf("maximum depth = %d\n", maxDepth(T));
    return 0;
}

TreeNode createTree()
{
    TreeNode T;
    int val;
     scanf("%d", &val);
     if (val != -1) {
        T = (TreeNode)malloc(sizeof(TNode));
        T->val = val;
        T->left = createTree();
        T->right = createTree();
     } else {
         T = NULL;
     }
     return T;
}

int maxDepth(TreeNode root)
{
    int l, r;
    if (NULL == root)
        return 0;
    l = maxDepth(root->left);
    r = maxDepth(root->right);
    return (l > r) ? l + 1 : r + 1;
}

File: LeetCode/MaximumDepthOfBinaryTree/maxDepthOfBT.py
#!/usr/bin/python
#-*- coding:utf-8 -*-

class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

class maxDepth:
    def createTree(self):
        val = input()
        if -1 == val:
            T = None
        else:
            T = TreeNode(val)
            T.left = self.createTree()
            T.right = self.createTree()
        return T

    def maxDepth(self, root):
        if None == root:
            return 0
        l = self.maxDepth(root.left)
        r = self.maxDepth(root.right)
        return (l + 1) if (l > r) else (r + 1)


if __name__=='__main__':
    m = maxDepth()
    T = m.createTree()
    print m.maxDepth(T)
